c/zss.c

```diff
@@ -847,12 +850,82 @@ static void readAgentAddressAndPort(JsonObject *serverConfig, JsonObject *envCon
   }
 }
 
+#define PORT_KEY         "port"
+#define IP_ADDRESSES_KEY "ipAddresses"
+#define KEYRING_KEY      "keyring"
+#define LABEL_KEY        "label"
+#define STASH_KEY        "stash"
+#define PASSWORD_KEY     "password"
+
+#define AGENT_HTTPS_PREFIX       "ZWED_agent_https_"
+#define ENV_AGENT_HTTPS_KEY(key) AGENT_HTTPS_PREFIX key
+
+static bool readAgentHttpsSettings(ShortLivedHeap *slh,
+                                   JsonObject *serverConfig,
+                                   JsonObject *envConfig,
+                                   char **outAddress,
+                                   int *outPort,
+                                   TlsSettings **outSettings
+                                  ) {
+  int port = jsonObjectGetNumber(envConfig, ENV_AGENT_HTTPS_KEY(PORT_KEY));
+  char *address = jsonObjectGetString(envConfig, ENV_AGENT_HTTPS_KEY(IP_ADDRESSES_KEY));
+
+  TlsSettings *settings = (TlsSettings*)SLHAlloc(slh, sizeof(*settings));
+  settings->keyring = jsonObjectGetString(envConfig, ENV_AGENT_HTTPS_KEY(KEYRING_KEY));
```

Comment:
does this only support keyrings?
can gskit also read in unix files?

Reply:
`keyring` is implicit, See for details https://github.com/zowe/zowe-common-c/blob/staging/h/tls.h#L26
I use a `.kdb` gskit database file, for example, see https://github.com/lchudinov/tls-examples